lib/request.js

```diff
@@ -242,7 +242,7 @@ module.exports = {
    */
 
   get hostname() {
-    const host = this.host;
+    const { host } = this;
```

Comment:
If top comment is enforced (preference `const proxy = this.app.proxy`) I would assume to find `const host = this.host` for both same reason as above and consistency throughout.